integration_setup.sh

```diff
@@ -146,26 +166,52 @@ create_storage_response=$(curl -X POST -u${OP_ADMIN_USERNAME}:${OP_ADMIN_PASSWOR
   -d @${INTEGRATION_SETUP_TEMP_DIR}/request_body_1_op_create_storage.json
 )
 if [[ $INTEGRATION_SETUP_DEBUG != "true"  ]]; then rm ${INTEGRATION_SETUP_TEMP_DIR}/request_body_1_op_create_storage.json; fi
-
 # check for errors
 response_type=$(echo $create_storage_response | jq -r "._type")
 if [[ ${response_type} == "Error" ]]; then
   error_message=$(echo $create_storage_response | jq -r ".message")
   error_id=$(echo $create_storage_response | jq -r ".errorIdentifier")
   if [[ ${error_id} == "urn:openproject-org:api:v3:errors:MultipleErrors" ]]; then
-    error_messages=$(echo $create_storage_response | jq -r "._embedded.errors[].message")
-    log_error "Got multiple errors when setting up
-      OP storage: ${OPENPROJECT_STORAGE_NAME}
-      for Nextcloud: '${NEXTCLOUD_HOST}
-      using endpoint: ${OPENPROJECT_BASEURL_FOR_STORAGE}"
-    log_error "Error Message(s): ${error_messages}"
-    log_info "You could try deleting the file storage in OpenProject and run the script again."
+    # if files storage is already created with the provided host and storage name
+    # we assume that the integration set up is done in both application
+    error_messages_grep=$(echo $create_storage_response | jq -r '.["_embedded"]["errors"] | .[].message')
+    readarray -t error_messages <<<"$error_messages_grep"
+    count=0
+    for element in "${error_messages[@]}"; do
+      if [[ "$element" == "Host has already been taken." ]]; then
+          (( count +=1 ))
+      elif [[ "$element" == "Name has already been taken." ]]; then
+          (( count +=1 ))
+      fi
+    done
+    if [[ "$count" -ne 2 ]]; then
```

Comment:
What about making sure that there are only 2 error messages and make the variable names more explicit?

```suggestion
    error_count = 0
    host_already_taken = false
    name_already_taken = false
    for element in "${error_messages[@]}"; do
      if [[ "$element" == "Host has already been taken." ]]; then
          host_already_taken = true
      elif [[ "$element" == "Name has already been taken." ]]; then
          name_already_taken = true
      fi
      (( error_count +=1 ))
    done
    if [[ $host_already_taken != true || $name_already_taken != true || "$error_count" -ne 2  ]]; then
```